apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx

```diff
@@ -1,11 +1,218 @@
+import Link from 'next/link';
+import { cn } from '@/lib/utils';
+import { useGlobalStore } from '@/providers/global-store-provider';
+import {
+  CircleUser,
+  Mail,
+  Lock,
+  Building2,
+  CreditCard,
+  AtSign,
+  Globe,
+  Users,
+  UserRoundPlus,
+  HeartHandshake
+} from 'lucide-react';
 
 export default function SettingsSidebar() {
+  const orgShortCode = useGlobalStore((state) => state.currentOrg.shortCode);
+  const pathname = usePathname();
+  const [currentUrl, setCurrentUrl] = useState('');
+
+  useEffect(() => {
+    setCurrentUrl(pathname);
+  }, [pathname]);
+
+  const personalLinks = [
+    {
+      key: `/${orgShortCode}/settings/user/profile`,
+      label: 'Profile',
+      to: `/${orgShortCode}/settings/user/profile`,
+      icon: <CircleUser />
+    },
+    {
+      key: `/${orgShortCode}/settings/user/addresses`,
+      label: 'Personal Addresses',
+      to: `/${orgShortCode}/settings/user/addresses`,
+      icon: <Mail />
+    },
+    {
+      key: `/${orgShortCode}/settings/user/security`,
+      label: 'Security',
+      to: `/${orgShortCode}/settings/user/security`,
+      icon: <Lock />
+    }
+  ];
+  const orgSetupLinks = [
+    {
+      key: `/${orgShortCode}/settings/org`,
+      label: 'Org Profile',
+      to: `/${orgShortCode}/settings/org`,
+      icon: <Building2 />
+    },
+    {
+      key: `/${orgShortCode}/settings/org/setup/billing`,
+      label: 'Billing',
+      to: `/${orgShortCode}/settings/org/setup/billing`,
+      icon: <CreditCard />
+    }
+  ];
+
+  const orgUserLinks = [
+    {
+      key: 'members',
+      label: 'Members',
+      to: `/${orgShortCode}/settings/org/users/members`,
+      icon: <Users />
+    },
+    {
+      key: `/${orgShortCode}/settings/org/users/invites`,
+      label: 'Invites',
+      to: `/${orgShortCode}/settings/org/users/invites`,
+      icon: <UserRoundPlus />
+    },
+    {
+      key: `/${orgShortCode}/settings/org/users/teams`,
+      label: 'Teams',
+      to: `/${orgShortCode}/settings/org/users/teams`,
+      icon: <HeartHandshake />
+    }
+  ];
+  const orgMailLinks = [
+    {
+      key: `/${orgShortCode}/settings/org/mail/domains`,
+      label: 'Domains',
+      to: `/${orgShortCode}/settings/org/mail/domains`,
+      icon: <Globe />
+    },
+    {
+      key: `/${orgShortCode}/settings/org/mail/addresses`,
+      label: 'Email Addresses',
+      to: `/${orgShortCode}/settings/org/mail/addresses`,
+      icon: <AtSign />
+    }
+  ];
+
   return (
-    <Flex className="bg-slate-2 dark:bg-slatedark-2 h-full w-[350px] p-2">
-      Settings Side Bar
+    <Flex
+      gap="4"
+      className="bg-slate-2 dark:bg-slatedark-2  h-full w-[400px] flex-col  p-2 px-4">
+      <Text
+        size="5"
+        weight="bold"
+        color="gray">
+        Personal
+      </Text>
+      <Flex
+        className="flex-col"
+        gap="1">
+        {personalLinks.map(({ key, label, to, icon }) => (
+          <Link
+            key={key}
+            href={to}>
```

Comment:
here you can do `key={to} href={to}`

Reply:
but shouldn't it better to have a separate key, improves readability, same goes for pathname

Reply:
there is no need to repeat the props, also you can just rename `pathname` as `currentUrl`

Reply:
Okk, doing it, but I still feel I am right

Reply:
using separate key is good when you don't know what your data might be, for this you are defining your keys yourself, so there is no chance of collision. Have to props named `key` and `to` with same string is not the best thing